src/Modules/SharedModule/components/SideBar/SideBar.tsx

```diff
@@ -1,7 +1,190 @@
+import ExitToAppIcon from '@mui/icons-material/ExitToApp';
+import GroupIcon from '@mui/icons-material/Group';
+import AppsIcon from '@mui/icons-material/Apps';
+import CalendarMonthIcon from '@mui/icons-material/CalendarMonth';
+import HomeIcon from '@mui/icons-material/Home';
+import MeetingRoomIcon from '@mui/icons-material/MeetingRoom';
+import BookmarkBorderIcon from '@mui/icons-material/BookmarkBorder';
 
 export default function SideBar() {
+  const navigate = useNavigate();
+  const { showToast } = useToast();
+  const [loading, setLoading] = useState(false);
+  const [collapsed, setIsCollapsed] = useState(true);
+  const [open, setOpen] = useState(false);
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+  let [collapsedWidth, setCollapsedWidth] = useState("80px");
+
+  const handleCollapse = ()=>{
+    setIsCollapsed(!collapsed)
+  }
+
+  const updateCollapsedWidth = () => {
+    const width = window.innerWidth;
+    if (width <= 576) {
+      setCollapsedWidth("60px");
+    } else if (width <= 768) {
+      setCollapsedWidth("80px");
+    } else if (width <= 992) {
+      setCollapsedWidth("80px");
+    } else {
+      setCollapsedWidth("80px");
+    }
+  };
+
+  useEffect(() => {
+    updateCollapsedWidth();
+    window.addEventListener('resize', updateCollapsedWidth);
+    return () => window.removeEventListener('resize', updateCollapsedWidth);
+  }, []);
+
+  const logout = ()=>{
+    setLoading(true);
+    setTimeout(() => {
+      localStorage.removeItem('token');
+      navigate('/login');
+      showToast("success", 'Logged out successfully');
+      setLoading(false);
+    }, 1000);
+  }
+  
   return (
-    <div>SideBar</div>
-  )
+    <>
+      <Modal
+        open={open}
+        onClose={handleClose}
+        aria-labelledby="modal-modal-title"
+        aria-describedby="modal-modal-description"
+      >
+        <Box
+          sx={{
+            width: 400,
+            bgcolor: "white",
+            position: "absolute",
+            top: "50%",
+            left: "50%",
+            transform: "translate(-50%, -50%)",
+            px: 3,
+            py: 5,
+            borderRadius: 2,
+          }}
+        >
+          <ChangePass />
+        </Box>
+      </Modal>
+
+      <div className="sidebar-container">
+        <Sidebar 
+          collapsed={collapsed} 
+          collapsedWidth={collapsedWidth}
+          >
+          <Menu style={{ backgroundColor: "orang", marginTop: 80 }}>
+            <MenuItem
+              onClick={handleCollapse}
+              style={{ textAlign: "center", marginBottom: 15 }}
+            >
+              <button
+                className={`collapse-button ${collapsed ? "" : "expanded"}`}
+              >
+                <ArrowForwardIcon
+                  sx={{ marginTop: 1, marginRight: { xs: 2, sm: 0 } }}
+                />
+              </button>
+            </MenuItem>
+
+            <MenuItem
```

Comment:
We can provide the paths and icons in an array, and then map over it to generate the items dynamically. This will make the code easier to follow and remove duplication of styles.